packages/orm/src/dataloaders/findDataLoader.ts

```diff
@@ -229,7 +233,8 @@ function buildConditions(ef: ParsedExpressionFilter, argsIndex: number = 0): [st
       argsIndex += argsTaken;
     }
   });
-  return [conditions.join(` ${ef.op.toUpperCase()} `), argsIndex];
```

Comment:
@allangaldinosilva you were exactly right that the bug was in `buildConditions`; this was my mistake, the return value here should be "the number of args taken" and not just "the latest index value".

Thanks for the great reproduction!